### PR TITLE
#1540 Fix autocomplete selector should use hooks api

### LINKED ISSUE #1540: AutocompleteSelector should use Hooks API
## The suggestion

AutocompleteSelector should use Hooks API rather than React Class API

**Why?**
Consistency and https://github.com/openmsupply/mobile/issues/1092

## Describe alternatives you've considered

Doing nothing.

### PR DESCRIPTION
**Pull request by sworup**
Fixes #1540

## Change summary

- Refactor code 
  - Autocomplete selector should use hooks api
  - Move `ResultRow` component out of `AutocompleteSelector` component
  - User `ResultRow` already present in code, hence sharing the component among `AutocompleteSelector` and `MultiSelectList`.

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Pages using `AutocompleteSelector` should work properly.
- [ ] Pages using `MultiSelectList` (Master-list selection) should work properly.
- [ ] Rest of the interface should be same. This is code refactor.

### Related areas to think about

None

### CODE REVIEW

File: src/widgets/AutocompleteSelector.js [hunk 1]
```diff
@@ -4,42 +4,37 @@
...
+    queryString,
+    queryStringSecondary,
+    primaryFilterProperty,
+    secondaryFilterProperty,
+    options,
+    onSelect,
+    placeholderText,
+    renderLeftText,
+    renderRightText,
+  } = props;
```
Comment (joshxg) [inline src/widgets/AutocompleteSelector.js]: could just destructure the parameter?

File: src/widgets/AutocompleteSelector.js [hunk 2]
```diff
@@ -77,50 +69,48 @@ export class AutocompleteSelector extends React.PureComponent {
...
+        color="white"
+        onChange={text => setQueryText(text)}
+        placeholder={placeholderText}
+        placeholderTextColor="white"
+        style={[localStyles.text, localStyles.searchBar]}
+      />
+      {data.length > 0 && (
+        <FlatList
+          data={data}
+          keyExtractor={item => item.id || item.name}
```
Comment (joshxg) [inline src/widgets/AutocompleteSelector.js]: could we define this explicitly so re-renders don't cause a new function to be created?

File: src/widgets/AutocompleteSelector.js [hunk 3]
```diff
@@ -77,50 +69,48 @@ export class AutocompleteSelector extends React.PureComponent {
...
+          renderItem={item => (
```
Comment (joshxg) [inline src/widgets/AutocompleteSelector.js]: same as above, can we memoize and explicitly define this callback to reduce re-renders?

File: src/widgets/ResultRow.js [hunk 4]
```diff
@@ -9,20 +9,21 @@ import { Text, TouchableOpacity, StyleSheet } from 'react-native';
 import Icon from 'react-native-vector-icons/Ionicons';
 import { APP_FONT_FAMILY, SUSSOL_ORANGE, BACKGROUND_COLOR, GREY } from '../globalStyles';
 
-const ResultRow = React.memo(
+export const ResultRow = React.memo(
   ({ data, renderLeftText, renderRightText, onPress, isSelected, showCheckIcon }) => {
-    const rowPressed = useCallback(() => onPress(data), [data]);
+    const rowPressed = useCallback(() => onPress(data.item), [data.item]);
```
Comment (joshxg) [inline src/widgets/ResultRow.js]: nice

File: src/widgets/index.js [hunk 5]
```diff
@@ -4,8 +4,9 @@
  */
 
 export { Button, ProgressBar } from 'react-native-ui-components';
-
+// eslint-disable-next-line import/no-cycle
```
Comment (joshxg) [inline src/widgets/index.js]: In the above files if we export directly from the files rather than from `.` we can remove the cycle, I believe

Comment (sworup) [reply src/widgets/index.js]: I thought about doing that, but did this for consistency. Will revert back.

File: src/widgets/AutocompleteSelector.js (hunk 2 above)
Comment (sworup) [reply src/widgets/AutocompleteSelector.js]: Resolved with 81441a9

File: src/widgets/AutocompleteSelector.js (hunk 3 above)
Comment (sworup) [reply src/widgets/AutocompleteSelector.js]: Resolved with 81441a9

File: src/widgets/AutocompleteSelector.js (hunk 1 above)
Comment (sworup) [reply src/widgets/AutocompleteSelector.js]: Resolved with ae8cb73

File: src/widgets/index.js (hunk 5 above)
Comment (sworup) [reply src/widgets/index.js]: Resolved with 7ae9c50

**Pull request merged**